### PR TITLE
move the call to init_prob_parameters() to after the Fortran read

### PR DESCRIPTION
**Pull request by zingale**
## PR checklist

- [ ] test suite needs to be run on this PR
- [ ] this PR will change answers in the test suite to more than roundoff level
- [ ] all newly-added functions have docstrings as per the coding conventions
- [ ] the `CHANGES` file has been updated, if appropriate
- [ ] if appropriate, this change is described in the docs

### CODE REVIEW

File: Source/driver/Castro.cpp [hunk 1]
```diff
@@ -598,9 +598,9 @@ Castro::Castro (Amr&            papa,
 
     buildMetrics();
 
-    // initialize the C++ values of the runtime parameters
     if (do_init_probparams == 0) {
```
Comment (maxpkatz) [inline Source/driver/Castro.cpp:601]: Well now this variable name needs to be updated.

Comment (zingale) [reply Source/driver/Castro.cpp]: done

**Comment by zingale** [comment]
this doesn't work properly if we modify one of the _prob_params values in the Fortran probinit,

in the current setup, we have:
* fortran initializes
* fortran probinit
* copy from Fortran to C++
* C++ probinit
* copy from C++ to Fortran
* ...
* initialize the state date

but now are proposing to do:
* fortran initializes
* copy from Fortran to C++
* fortran probinit
* C++ probinit
* copy from C++ to Fortran
* ...
* initialize the state date

so if the fortran probinit changes the value of a variable in the `_prob_init`, then it is not changed in C++ and we overwrite it when we copy from C++ to Fortran.

**Comment by zingale** [comment]
now that we don't have any Fortran probinit that can change values, I think that this PR is okay.  This will then support reading the problem parameters in C++ directly (#1532)

**Comment by zingale** [comment]
tests: http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2021-05-22-006/index.html

**Review by maxpkatz — APPROVED** [review]

**Pull request merged**